### ISSUE TITLE
Do we need delete the "softmax" before using sparse_categorical_crossentrop?

### ISSUE THREAD

**Issue by godissaw** [fchollet/keras#6587]
As we all know, the softmax activation layer output the same filters with the input.
I am doing the segmentation tasks.
For example, use the voc FCN, the  second-last is (none, nb_classes=21, w, h)
after the softmax, it output the same (none, nb_classes=21, w, h),
So the y_pred =(21,w,h),but the y_ture =(1,w,h) ,if I don't want to  one-hot every dot in my label, I think the "sparse_categorical_crossentrop" works.
However, when I check the loss function, I found that it bases on this "tf.nn.sparse_softmax_cross_entropy_with_logits" (tf backend), it has the softmax function.
So, should I delete the softmax before I using  it?

**Comment by godissaw** [comment]
I tried to delete it， then it did not fit. So it means we need the softmax layer.
But why?

**Comment by t-ae** [comment]
`sparse_categorical_crossentropy` loss has `from_logits` argument, and it is `False` in  default.
https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py#L1505
